synapse/replication/http/send_event.py

```diff
@@ -92,8 +112,20 @@ def __init__(self, hs):
         self.store = hs.get_datastore()
         self.clock = hs.get_clock()
 
+        # The responses are tiny, so we may as well cache them for a while
+        self.response_cache = ResponseCache(hs, timeout_ms=30 * 60 * 1000)
+
+    def on_PUT(self, request, event_id):
+        result = self.response_cache.get(event_id)
+        if not result:
```

Comment:
I'd suggest adding a line of logging for the case where there is an entry in the cache already.